crates/ruff_linter/src/checkers/noqa.rs

```diff
@@ -128,7 +128,13 @@ pub(crate) fn check_noqa(
                         }
 
                         if line.matches.iter().any(|match_| *match_ == code)
+                            // Perform a fast-check for exact matches first
                             || settings.external.contains(code)
+                            // then check for matching prefixes
+                            || settings
+                                .external
+                                .iter()
+                                .any(|external| code.starts_with(external))
```

Comment:
Since most codes _won't_ match an external, I bet it's better to just do this iteration every time and skip the direct check, but obviously can't say for sure.

It's probably even better to just use a vector instead of a hash set, since we're doing this iteration every time, and the set of externals is gonna tend to be quite small.

Reply:
(Defer to you as to whether you want to change.)

Reply:
Okee